### ISSUE TITLE
Add possibility to exclude last sample from methods Trace.slice(), Trace.trim() 

### ISSUE THREAD

**Issue by petrrr** [obspy/obspy#1288]
Here I am proposing to add a parameter to the two methods, which would allow to exclude the last limit sample of the selected time window. This would be sometimes quite useful when iteration over consecutive time windows, details matter and one would like to avoid including the limiting sample twice. 

I do not see an elegant workaround, and think adding this little feature would be quite useful to others. 

Not sure if there might be interest in excluding the starting limit sample as well would be interesting as well, maybe for "symmetry"? Probably this hypothetical parameter should apply to `Trace.slide()` method as well. 

Any comments?

**Comment by krischer** [comment]
One just has to be careful what happens when the endtime is exactly at a sample. Also how this works with the `nearest_sample` parameter should be discussed. 

By and large and I think this is a good idea - the parameter could be called `exclude_last_sample` and should of course default to `False`.

**Comment by petrrr** [comment]
I guess, there should be something well defined (or controllable) tolerance, where the sample would be regarded aligned with the limit. Otherwise the sample falls either inside or outside the intervall.

Okay, will try to look into this ASAP, not sure when.

**Comment by megies** [comment]
I've also come across a similar use case now..
For me it would be interesting to have an option to leave away the last sample conditionally, if (and only if) the last sample would match exatly the given time, i.e. if..
```
part1 = tr.slice(endtime=t)
part2 = tr.slice(starttime=t)
```
..would lead to one sample being part of both slices.

**Comment by Jollyfant** [comment]
In the QC module I always subtract 1μs from the endtime and use `nearest_sample=False` to make sure I'm not including a sample located on the endtime.

**Comment by megies** [comment]
Yeah, something like that works too of course, but it's a bit hacky, and looks a bit cryptic in the code to read, so we still might want to think about adding some API for this.
